packages/log/src/Logger.ts

```diff
@@ -0,0 +1,66 @@
+/* eslint no-console: "off" */
+import { ERROR, WARN, INFO, DEBUG, DEBUG2 } from './LoggerLevel';
+
+const silent = () => undefined;
+
+class Logger {
+  constructor(name: string | null, level: number | undefined) {
+    this.name = name;
+    this.prefix = name ? `[${name}]` : ``;
+
+    this.error = silent;
+    this.warn = silent;
+    this.log = silent;
+    this.info = silent;
+    this.debug = silent;
+    this.debug2 = silent;
+
+    this.setLogLevel(level);
+  }
+
+  name: string | null;
+
+  prefix: string;
+
+  level?: number;
+
+  error: (...data: unknown[]) => void;
+
+  warn: (...data: unknown[]) => void;
+
+  log: (...data: unknown[]) => void;
+
+  info: (...data: unknown[]) => void;
+
+  debug: (...data: unknown[]) => void;
+
+  debug2: (...data: unknown[]) => void;
+
+  setLogLevel(level?: number): void {
+    this.level = level;
+
+    this.error =
+      level != null && ERROR <= level
```

Comment:
Minor comment that I think helps the readability of the code and is a good thing to think about when doing comparisons (I'm aware you were just copying this logic here)

When comparing to a constant, put the variable first in the comparison. It's minor, but helps the readability. I think most people would say "if the log level is greater or equal to the error level" which corresponds to `level >= ERROR` and not "if the error level is less or equal to the current level" like written

Functionally the same, but makes the code just ever so slightly easier to read. For these cases it's not complicated really, but if you have 4 or 6 boolean checks in a condition, it can make the conditional easier to understand